spec/imsc-hrm.html

```diff
@@ -686,14 +681,17 @@ <h2>Paint Images</h2>
 
       <p>NSIZ(I<sub>i</sub>) is the number of pixels of <a>presented image</a> I<sub>i</sub>.</p>
 
-      <p>The contents of the Decoded Image Buffer D<sub>n</sub> are transferred instantaneously to Decoded Image Buffer
-      D<sub>n-1</sub> at the presentation time of <a>Intermediate Synchronic Document</a> E<sub>n</sub>.</p>
+      <p>At the presentation time of <a>Intermediate Synchronic Document</a> E<sub>n</sub>:</p>
+      <ol>
+        <li>purge from the Decoded Image Cache all images not flagged <i>retain</i>; and</li>
+        <li>remove the <i>retain</i> flag from all remaining images; and</li>
+      </ol>
 
-      <p>The total size occupied by images stored in Decoded Image Buffers D<sub>n</sub> or D<sub>n-1</sub> is the sum of
-      their Normalized Image Area.</p>
+      <p>It SHALL be an <a>error</a> if the sum of Normalized Image Area images stored in Decoded Image Cache is
+      greater than the Normalized Decoded Image Cache Size (NDIBS).</p>
```

Comment:
At what point in the process is this potential error condition checked? Would it not make sense to check which images are being reused, discard the remainder, and _then_ attempt to decode the new image(s), and only then check it the error condition has been met? As it stands, all new images are decoded before discarding images that are not being reused.